src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/commissioner/SelectNetworkFragment.java

```diff
@@ -0,0 +1,126 @@
+import androidx.navigation.NavController;
+import androidx.navigation.fragment.NavHostFragment;
+import com.google.chip.chiptool.R;
+import com.google.chip.chiptool.setuppayloadscanner.CHIPDeviceInfo;
+
+public class SelectNetworkFragment extends Fragment {
+
+  private CHIPDeviceInfo deviceInfo;
+
+  private NetworkAdapter networksAdapter;
+
+  private NetworkInfo selectedNetwork;
+  private Button addDeviceButton;
+
+  private BorderAgentDiscoverer borderAgentDiscoverer;
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    networksAdapter = new NetworkAdapter(getContext());
+    borderAgentDiscoverer = new BorderAgentDiscoverer(getContext(), networksAdapter);
+    borderAgentDiscoverer.start();
+  }
+
+  @Override
+  public void onDestroy() {
+    super.onDestroy();
+
+    borderAgentDiscoverer.stop();
+  }
+
+  @Override
+  public View onCreateView(
+      LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    // Inflate the layout for this fragment
+    return inflater.inflate(R.layout.commissioner_select_network_fragment, container, false);
+  }
+
+  public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+
+    // Hide the button
+    addDeviceButton = view.findViewById(R.id.add_device_button);
+    addDeviceButton.setVisibility(View.GONE);
+
+    deviceInfo = getArguments().getParcelable(Constants.KEY_DEVICE_INFO);
+
+    String deviceInfoString =
+        String.format(
+            "version: %d\nvendorId: %d\nproductId: %d\nsetupPinCode: %d",
+            deviceInfo.getVersion(),
+            deviceInfo.getVendorId(),
+            deviceInfo.getProductId(),
+            deviceInfo.getSetupPinCode());
+    TextView deviceInfoView = view.findViewById(R.id.device_info);
+    deviceInfoView.setText(deviceInfoString);
+
+    final ListView networkListView = view.findViewById(R.id.networks);
+    networkListView.setAdapter(networksAdapter);
+
+    networkListView.setOnItemClickListener(
+        new AdapterView.OnItemClickListener() {
+          @Override
+          public void onItemClick(AdapterView<?> adapterView, View view, int position, long id) {
+            selectedNetwork = (NetworkInfo) adapterView.getItemAtPosition(position);
+            addDeviceButton.setVisibility(View.VISIBLE);
+          }
+        });
+
+    view.findViewById(R.id.add_device_button)
+        .setOnClickListener(
+            new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                NavController controller =
+                    NavHostFragment.findNavController(SelectNetworkFragment.this);
+
+                if (controller.getCurrentDestination().getId()
+                    == R.id.commissioner_select_network_fragment) {
+                  Bundle bundle = new Bundle();
+
+                  assert (deviceInfo != null);
```

Comment:
Please consider using Log.e to log the issue when deviceInfo or selectedNetwork is null instead of using assert. It will help explain the issue instead of just crashing the app.

Reply:
updated.